src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -245,6 +245,11 @@ public int SelectedIndex
 
 		private static object CoerceSelectedIndex(DependencyObject dependencyObject, object baseValue)
 		{
+			if (baseValue == null)
```

Comment:
We should probably be checking for baseValue being an integer, it will avoid dealing with equality comparers of the provided instance, if any.

Reply:
Fixed!